tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackageRunTestCase.java

```diff
@@ -0,0 +1,114 @@
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+/**
+ * Integration test cases for running packages.
+ */
+public class PackageRunTestCase extends BaseTest {
+    private Path tempProjectDirectory;
+    private Map<String, String> envVariables;
+
+    @BeforeClass()
+    public void setUp() throws IOException {
+        tempProjectDirectory = Files.createTempDirectory("bal-test-integration-run-package-project-");
+        envVariables = PackagingTestUtils.getEnvVariables();
+    }
+
+    /**
+     * Running an empty package in a project.
+     *
+     * @throws BallerinaTestException When an error occurs executing the command.
+     */
+    @Test(description = "Test running an empty package")
+    public void testRunWithEmptyPkg() throws BallerinaTestException, IOException {
+        Path projectPath = tempProjectDirectory.resolve("firstTestProject");
+        initProject(projectPath);
+
+        // Create empty directory
+        Files.createDirectories(projectPath.resolve("emptypkg"));
+
+        LogLeecher clientLeecher = new LogLeecher("error: no ballerina source files found in package emptypkg");
+        balClient.runMain("run", new String[]{"emptypkg"}, envVariables, new String[0],
+                          new LogLeecher[]{clientLeecher}, projectPath.toString());
+        clientLeecher.waitForText(3000);
+    }
+
+    /**
+     * Running a package with text files in a project.
+     *
+     * @throws BallerinaTestException When an error occurs executing the command.
+     */
+    @Test(description = "Test running a package with text files")
+    public void testRunWithOtherPkg() throws BallerinaTestException, IOException {
+        Path projectPath = tempProjectDirectory.resolve("secondTestProject");
+        initProject(projectPath);
+
+        // Create empty directory
+        createDirWithTextFile(projectPath);
+
+        LogLeecher clientLeecher = new LogLeecher("error: no ballerina source files found in package otherpkg");
+        balClient.runMain("run", new String[]{"otherpkg"}, envVariables, new String[0],
+                          new LogLeecher[]{clientLeecher}, projectPath.toString());
+        clientLeecher.waitForText(3000);
+    }
+
+    /**
+     * Init project.
+     *
+     * @param projectPath project path
+     * @throws IOException            if an I/O exception occurs when creating directories
+     * @throws BallerinaTestException
+     */
+    private void initProject(Path projectPath) throws IOException, BallerinaTestException {
+        Files.createDirectories(projectPath);
+        String[] options = {"\n", "integrationtests\n", "\n", "f\n"};
+        balClient.runMain("init", new String[]{"-i"}, envVariables, options, new LogLeecher[0],
+                          projectPath.toString());
+    }
+
+    /**
+     * Create directory with a text file.
+     *
+     * @param path path of the directory to be created
+     * @throws IOException
```

Comment:
Shall we add description here? Check all the places.

Reply:
Added descriptions in all the places with 5d30629